ext/vscode/package.json

```diff
@@ -464,6 +464,9 @@
             }
         ]
     },
+    "extensionDependencies": [
+        "ms-vscode.azure-account"
```

Comment:
The Azure Account extension is headed toward deprecation so we should use the built-in Microsoft authentication provider instead.

Reply:
@bwateratmsft - Good to know, I wasn't exactly sure what to integrate with and this was the first thing I fond :-). Just confirming - It looks like we should instead be using https://code.visualstudio.com/api/references/vscode-api#authentication with the `'microsoft'` authentication provider, correct?

Do you know if there's an existing package that exposes this stuff in a [`TokenCredential`](https://learn.microsoft.com/en-us/javascript/api/@azure/core-auth/tokencredential?view=azure-node-latest) like all the Azure SDKs expect as the `ms-vscode.azure-account` did?  If no, I can write a wrapper, just wanted to re-use some existing package or logic someone else might have already written.

Reply:
Funny you should ask :wink: : https://github.com/microsoft/vscode-azuretools/tree/main/auth

Reply:
Fundamentally, that package is doing what the Azure Account extension currently does--list subscriptions and attach a token for each subscription. You may be able to skip the subscription listing part though, going straight to making a `TokenCredential` and saving some lookup time.

Reply:
I ending up not using `vscode-azext-azureauth` for now and instead just pulling in some of the code here with some TODOs.  The major problem I was running into is around support for an account which is granted access to multiple tenants (my MSA is one such account, because we have a test tenant that @vhvb1989 set up for us for some testing and I am also a member of that tenant).

I was running into cases where `isLoggedIn()` would return true (because I was logged into the default tenant) but calling `getSubscriptions()` would fail because it would try to fetch information from the tenant I was not authenticated to.

With these changes, there is still some weirdness in this case, because `azd`, when trying to list all the subscriptions you have access to will end up making a Subscriptions Client per tenant and try to list all the subscriptions the user has access to.  That will cause a login request per tenant in VS Code (which means you may end up with multiple prompts from the VS Code saying the Azure Developer Extension wants to use your authentication information and ask you to log in to each tenant) which is sort of a weird experience.

Perhaps we want to also delegate subscription listing to this endpoint.  That would also allow us to respect any filters the user had picked in VS Code, since the extension could rendezvous with the rest of VS Code to get a list of filtered subscriptions.

Also, there's that issue we discussed about how the token returned by the existing `vscode-azext-azureauth` extension has an expiresOnTimestamp of 0, because the VS Code authentication system doesn't return this value when it gives you back an authentication session.  We discussed opportunistically pulling the `exp` claim, as I've done here, which we don't yet do for the `vscode-azext-azureauth` because you want that extension to be isomorphic and the browser and client disagree on how you spell "base64 decode this string".

Talking with folks on the Azure Identity team, it seems like the outcome of handing back a token that "appears" to have expired like this will just mean it won't get cached, instead of being rejected, so I think we'd be fine either way, but since the logic is in the `azure-dev` extension right now (which is doing a bunch of stuff that won't work in the browser, like spawning process (and now launching HTTP servers), I did the work to crack the JWT and use the claim if it was there.